### ISSUE TITLE
@testitem blocks should have isolated scopes to avoid false InvalidRedefofConst warnings

### ISSUE THREAD

**Issue by cuihantao** [julia-vscode/StaticLint.jl#405]
Problem When a test file contains multiple @testitem blocks (from TestItemRunner.jl), each with the same using statements, StaticLint reports false "Invalid redefinition of constant" warnings. Example:

```julia
using TestItems

@testitem "Test A" begin
    using SomePackage  # <-- Warning: Invalid redefinition of constant
    # ...
end

@testitem "Test B" begin
    using SomePackage  # <-- Warning: Invalid redefinition of constant
    # ...
end
```

**Root Cause** 

In scope.jl, the introduces_scope function does not include :macrocall as a scope-introducing expression. This means @testitem blocks share the file-level scope, causing duplicate using statements to be flagged as redefinitions. At runtime, each @testitem block executes in an isolated module, so these using statements are valid and don't conflict. 

**Proposed Fix** 

Add special handling for @testitem (and @testset) in macros.jl, similar to how @eval is handled:

```julia
elseif _is_testitem_macro(x.args[1])
    # @testitem and @testset blocks should have their own scope
    # to avoid false "Invalid redefinition of constant" errors
    # when multiple testitems in the same file use the same bindings
    setscope!(x, Scope(x))
    setparent!(scopeof(x), state.scope)
    s0 = state.scope
    state.scope = scopeof(x)
    for i = 3:length(x.args)
        state(x.args[i])
    end
    state.scope = s0
    return # Already processed children
```
With helper:

```
function _is_testitem_macro(x)
    isidentifier(x) || return false
    name = valofid(x)
    return name == "@testitem" || name == "@testset"
end
```

**Note on Local Testing** 

Testing this fix locally with VS Code is difficult because I don't know how to wire this local dev repo to Julia Language Server. Even after:
Running (in base env)
```
Pkg.develop(path="path/to/StaticLint.jl")
```

Clearing `~/.julia/compiled/*/StaticLint` and `~/.julia/compiled/*/LanguageServer`

Restarting the language server...

the language server may still use a cached or bundled version. There's no clear way to verify which StaticLint version the LS is actually using. Any help would be greatly appreciated.

**Comment by cuihantao** [comment]
A similar (but slightly different) issue is in https://github.com/julia-vscode/StaticLint.jl/issues/401.

**Comment by NIKHIL-cell587** [comment]
Hi everyone, I am Nikhil Gupta (@NIKHIL-cell587). I have submitted my GSoC 2026 proposal for the project 'Static Analysis and Linting for Julia'. My focus is on improving the developer experience by enhancing real-time error detection and code quality tools.Although I haven't made major contributions yet, I have deeply studied the StaticLint.jl and LanguageServer.jl codebases as outlined in my roadmap. I am eager to start with some 'good first issues' or help in improving documentation to get familiar with the workflow. Could the mentors point me towards a small task that aligns with the current goals of the Julia VS Code extension?
